flymake-vale.el

```diff
@@ -139,6 +142,19 @@ Passing the results and source BUF to CALLBACK."
   (or  (string-equal event "finished\n")
        (string-match "exited abnormally with code 1.*" event)))
 
+(defun flymake-vale--detect-extension ()
+  "Attempt to detect a file extension related to the buffer we wish to
+check, either using the file extension or the major mode."
+  (with-current-buffer flymake-vale--source-buffer
+    (let ((file-name (buffer-file-name)))
+      (cond ((not (string-empty-p 'flymake-vale-extension)) `("--ext" ,flymake-vale-extension)
+             (not (null file-name)) `("--ext" ,(file-name-extension file-name t)))))))
```

Comment:
Rewriting this without using `cond` would be a bit clearer,  I'd suggest something like this: 

```
(let* ((f (buffer-file-name flymake-vale--source-buffer))
         (ext (or flymake-vale-file-ext (and f (file-name-extension f t)))))
    (list (if ext (concat "--ext=" ext) "")))
```